miniweb-core/src/main/java/com/devsmart/miniweb/Server.java

```diff
@@ -146,35 +197,56 @@ public void run() {
             httpService.setHandlerResolver(requestHandlerResolver);
             httpService.setParams(params);
 
-            Socket socket = null;
+            Socket rawSocket = null;
             while (mRunning) {
                 try {
                     if (mIsDebugBuild) {
                         LOGGER.info("waiting in accept on {}", mServerSocket.getLocalSocketAddress());
                     }
-                    socket = mServerSocket.accept();
+                    rawSocket = mServerSocket.accept();
                     if (mIsDebugBuild) {
-                        LOGGER.info("accepting connection from: {}", socket.getRemoteSocketAddress());
+                        LOGGER.info("accepting connection from: {}", rawSocket.getRemoteSocketAddress());
+                    }
+                    Socket boundSocket = rawSocket;
+
+                    // If SSL is enabled, ensure we have a SSLSocket and complete the handshake
+                    if (mSslEnabled && (mServerSocket instanceof SSLServerSocket)) {
+                        SSLSocket sslSocket = (SSLSocket) rawSocket;
+                        sslSocket.setUseClientMode(false);
+                        sslSocket.setEnabledProtocols(new String[]{"TLSv1.2"});
+                        sslSocket.startHandshake();
                     }
 
                     DefaultHttpServerConnection connection = new DefaultHttpServerConnection();
-                    connection.bind(socket, new BasicHttpParams());
-                    RemoteConnection remoteConnection = new RemoteConnection(socket.getInetAddress(), connection);
+                    connection.bind(boundSocket, new BasicHttpParams());
+                    RemoteConnection remoteConnection = new RemoteConnection(boundSocket.getInetAddress(), connection);
 
                     mWorkerThreads.execute(new WorkerTask(httpService, remoteConnection));
+                } catch (SSLHandshakeException e) {
+                    LOGGER.error("SSL Handshake failed: {}. Reverting to HTTP.", e.getMessage());
+                    try {
+                        rawSocket.close();
+                        LOGGER.info("Connection is closed properly");
+                    } catch (IOException ioException) {
+                        LOGGER.error("Can't close connection. Reason: ", ioException);
+                    }
+                    fallbackToHttp();
```

Comment:
HI @AnkitManeSciAps - Is this just for testing and experimenting?  I'm not sure if we'll need or want this on the analyzer.  We'll definitely need it on the client (PB) side.